src/_igraph/graphobject.c

```diff
@@ -7740,6 +7740,104 @@ PyObject *igraphmodule_Graph_layout_sugiyama(
 
   igraph_vector_int_destroy(&extd_to_orig_eids);
   return (PyObject *) result_o;
+
+
+  /** \ingroup python_interface_graph
+ * \brief Places the vertices of a graph using Uniform Manifold Approximation and Projection (UMAP)
+ * \return the calculated coordinates as a Python list of lists
+ * \sa igraph_layout_umap
+ */
+PyObject *igraphmodule_Graph_layout_umap(igraphmodule_GraphObject * self,
+                                        PyObject * args, PyObject * kwds)
+{
+  static char *kwlist[] =
+    { "dist", "dim", "min_dist", "epochs", "sampling_prob", NULL };
+  igraph_matrix_t m;
+  igraph_matrix_t *dist = 0;
+  /* FIXME: use reasonable defaults */
+  double min_dist = 0.01, sampling_prob = 0.5;
+  Py_ssize_t dim = 2, Py_ssize_t epochs = 50;
+  PyObject *dist_o = Py_None;
+  PyObject *result_o;
+
+  /* arpack_options_o is now unused but we kept here for sake of backwards
+   * compatibility */
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "|Ondnd", kwlist, &dist_o,
+                                   &dim, &min_dist, &epochs, &sampling_prob))
+    return NULL;
+
+  CHECK_SSIZE_T_RANGE_POSITIVE(dim, "number of dimensions");
+  CHECK_SSIZE_T_RANGE_POSITIVE(epochs, "number of epochs");
+
+  if (min_dist < 0) {
+    /* FIXME: How to actually tell the user? */
+    igraphmodule_handle_igraph_error();
+  }
+
+  if (sampling_prob <= 0) {
+    /* FIXME: How to actually tell the user? */
+    igraphmodule_handle_igraph_error();
+  }
+
+  if (sampling_prob > 1) {
+    /* FIXME: How to actually tell the user? */
+    igraphmodule_handle_igraph_error();
+  }
```

Comment:
These kinds of checks are normally done only in the C core and not duplicated in the high-level interface.

Reply:
Thanks. What about `CHECK_SSIZE_T_RANGE_POSITIVE` the lines right above?

Reply:
Okay I removed all of that, it's checked later on anyway.

Reply:
`CHECK_SSIZE_T_RANGE_POSITIVE` is Python-specific so it should stay. It checks that the integer coming from Python fits into an `igraph_integer_t`.

To be clear, I'm not at all experienced with the Python interface, and part of the reason I'm looking at this is to better understand how to expose functions to Python.